### PR TITLE
Fix for TN-602, TN-603 and TN-604  implement access_on_verify role

### PR DESCRIPTION
**Pull request by pbugni**
Fix for TN-602, TN-603 and TN-604 - namely, full implementation for MUSIC's needs of an access_on_verify role.

### CODE REVIEW

File: portal/migrations/versions/124cffb0fc6f_.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+
+def downgrade():
+    # Create old type for restoration - need tmp for storage
+    tmp_type.create(op.get_bind(), checkfirst=False)
+    op.execute(
+        'ALTER TABLE encounters ALTER COLUMN auth_method TYPE _auth_methods'
+        ' USING auth_method::text::_auth_methods')
+    new_type.drop(op.get_bind(), checkfirst=False)
+
```
Comment (OptimusRhine) [inline portal/migrations/versions/124cffb0fc6f_.py]: Do we need to convert the `auth_method` values of any Encounters using the new 'url_authenticated_and_verified' value to one of our old values (maybe 'url_authenticated') before fully converting the column back to the old Enum?

File: portal/views/user.py [hunk 2]
```diff
@@ -372,9 +372,9 @@ def access_url(user_id):
     if not has.isdisjoint(not_allowed):
         abort(400, "Access URL not provided for privileged accounts")
 
-    if ROLE.WRITE_ONLY not in has:
+    if set((ROLE.ACCESS_ON_VERIFY, ROLE.WRITE_ONLY)).isdisjoint(has):
         # KEEP this restriction.  Weak authentication (which the
-        # returned URL provides) should only be available for WRITE_ONLY users
+        # returned URL provides) should only be available for these roles
         abort(400, "Account {} lacks WRITE_ONLY role".format(user_id))
```
Comment (OptimusRhine) [inline portal/views/user.py]: Should this abort message reference the 'ACCESS_ON_VERIFY' role?

File: portal/migrations/versions/124cffb0fc6f_.py (hunk 1 above)
Comment (pbugni) [reply portal/migrations/versions/124cffb0fc6f_.py]: well - the encounter type didn't exist before.  i only would expect to rollback if a mistake was made, but I guess you're right, they could have come into existence, and then we noticed the need to rollback...

File: portal/views/user.py (hunk 2 above)
Comment (pbugni) [reply portal/views/user.py]: yup!

**Pull request merged**